project/app/api/covid_score.py

```diff
@@ -0,0 +1,204 @@
+import os
+import re
+
+import psycopg2
+import datetime
+from datetime import datetime, timedelta
+
+from app.api.constants import STATE_POP
+
+# Return Covid data for a given state and date from the database
+def req_cdc_covid_dat(db_conn, ste, dte):
+    """
+    req_cdc_covid_dat return Covid data for a given state and date
+    from the backend data science database
+
+    Parameters:
+        "db_conn":  database connection object
+        "ste":      the targeted state (e.g. "CA")
+        "dte":      the string date of the covid data requested (format: YYYY-MM-DD)
+
+    Returns:
+        "ok":       boolean indicating a successful request
+        "new_case": new cases logged in the API response
+        "data":     copy of the CDC API response
+        "error":    error message (if applicable)  
+    """
+    # Define a return object
+    ret_dict = {
+        "ok":         False,
+        "new_case":   None,
+        "data":       None,
+        "error":      "an error occurred"
+    }
+
+    # Do we have a valid state value?
+    if ste not in STATE_POP:
+        # invalid state parameter
+        ret_dict["error"] = "invalid state parameter: " + ste
+        return ret_dict
+
+    # Do we have a valid date value?
+    if re.match(r"^\d{4}\-\d{2}\-\d{2}$", dte) == None:
+        # dte parameter value is not valid
+        ret_dict["error"] = "invalid date parameter: " + dte
+        return ret_dict
+
+    # Query the database
+    sql = "SELECT * FROM state_covid_stats WHERE state = %s AND date = %s"
+    cvd_dict = {}
+    try:
+        cursor = db_conn.cursor()
+        cursor.execute(sql, (ste, dte))
+        cvd_row = cursor.fetchone()
+        cvd_dict["new_case"] = cvd_row[3]
+        cvd_dict["data"]     = cvd_row[5]
+
+    except (Exception, psycopg2.Error) as error:
+        ret_dict["error"] = f"error fetching covid data for state: {ste} date: {dte} - {error}"
+        return ret_dict
+    
+    # Return results
+    ret_dict["ok"]       = True
+    ret_dict["error"]    = None
+    ret_dict["new_case"] = cvd_dict["new_case"]
+    ret_dict["data"]     = cvd_dict["data"]
+    return ret_dict
+
+# Generate a Covid "score" for a given state
+def gen_covid_score(db_conn, ste):
```

Comment:
Great work get the calculations for the covid flags. Would suggest breaking this function into smaller, more modular subfunctions for ease of testing/updating.